### ISSUE TITLE
Add action to add top-level zettel

### ISSUE THREAD

**Issue by jamesx4** [Dyldog/luhman-obsidian-plugin#20]
We have ability to create a child or sibling note which requires you to navigate to an existing note to use as context.  Would love to be able to create hotkey that creates a new top-level note in my ZettelKasten starting with the next top-level id.  

So glad to see new activity starting up on this plugin!!

**Comment by brannonh** [comment]
Will need to consider #14 with this option.

**Comment by Eforen** [comment]
I think this will be a very good entry point when one has no top level zettels. I noticed that this can be daunting for newbies.

**Comment by jamesx4** [comment]
Even if you have some top level zettels, this could be handy to start up a whole new line of thought.  It may not be a sibling of another top level zettel thus doesn't need the link between zettels that Create Sibling gives you.

**Comment by Eforen** [comment]
oh that makes sense to have both options good point I will keep that in mind

I think we should have 2 commands:
- Create new root zettel
- Create new root sibling

This may mean that we have to do something weird somewhere

**Comment by brannonh** [comment]
Wouldn't the create sibling command work the same on a top-level zettel? We would just need a create root zettel command, right?

**Comment by Eforen** [comment]
> Wouldn't the create sibling command work the same on a top-level zettel? We would just need a create root zettel command, right?

It would but you have to be in the root Zettle to do that a create root sib command would make it so you can make a new root sibling if you are anywhere.

I don't personally have a usecase for eather of these though I feel some people would.

**Comment by brannonh** [comment]
Makes sense. I would almost call the command something like `Create New Zettel` in that case, since it's creating at the root. "Create new root sibling" is a bit ambiguous.

**Comment by brannonh** [comment]
> I think we should have 2 commands:
> 
> Create new root zettel
> Create new root sibling

I just reread this part. I think we can get away with a single command. Creating a new root sibling is essentially the same as creating a new root zettel, isn't it?

**Comment by Eforen** [comment]
True maybe along with this feature we merge the boxes together and put a drop down or something that can be changed to say where to make it maybe even give a filename preview. Sorta like the checkbox I added to the new child/sibling options. We should still have a dif command for each though so people can assign their own shortcuts to the dif commands if they want.

**Comment by Eforen** [comment]
> > I think we should have 2 commands:
> > 
> > Create new root zettel
> > Create new root sibling
> 
> I just reread this part. I think we can get away with a single command. Creating a new root sibling is essentially the same as creating a new root zettel, as far as _what_ is actually created, isn't it?

No the new root sibling command would be the same as going into the `1` note and running the new sibling command where as the new root note/zettel command would make a new `1` note in another folder or something thought if we do this we would need to have some consept of tree classification because how does it know which Zettelkasten it's in if people have multiple?

**Comment by brannonh** [comment]
Does this plugin even support multiple Zettelkasten currently? I didn't think it did.

**Comment by Eforen** [comment]
> Does this plugin even support multiple Zettelkasten currently? I didn't think it did.

No it doesn't seem to work that way. We would need to make a mode to support that. Maybe something like what folder things are in or something not sure

**Comment by TheHuntyBadger** [comment]
I use QuickAdd to create a new Top Level Zettel. I have a Zettel template and use the following to generate a new top level ID. The plug in does respect the Top level Zettels I create this way, 

I have to paste the ID that's created when I create the new Zettel into the Title,

This creates an ID of #$#$, you could shorten it to #$ but you would then be limited on the number of top level Zettels.

`## ID: <% Math.floor((Math.random()*10)+0).toString(16) %><% Math.floor((Math.random()*(16-10+1))+10).toString(16) %><% Math.floor((Math.random()*10)+0).toString(16) %><% Math.floor((Math.random()*(16-10+1))+10).toString(16) %>`

**Comment by TheHuntyBadger** [comment]
After posting this I got to thinking another way to do this. I created a script to run using Templater & QuickAdd to create a top level Zettel.

[Gist for Creating Luhmann style IDs of custom length](https://gist.github.com/TheHuntyBadger/723e09cd05e55811b9c53b67ee152051)

**Comment by brunops** [comment]
Hi, newbie here. How do I even create the first Zettel for any command to work? Tried to manually add numbers to titles like `1 test` or `1. test` etc, but still get and error.
![image](https://github.com/Dyldog/luhman-obsidian-plugin/assets/1735844/f7ce8762-dff0-4a70-b9f7-7d8c7b784329)

Never mind, was missing the separator
